pnut.c

```diff
@@ -728,21 +729,28 @@ char *file_parent_directory(char *path) {
   return path;
 }
 
-void include_file(char *file_name, char *relative_to) {
+FILE *fopen_source_file(char *file_name, char *relative_to) {
+  FILE *fp;
+  fp_filepath = file_name;
   if (relative_to) {
-    file_name = str_concat(relative_to, file_name);
+    fp_filepath = str_concat(relative_to, fp_filepath);
   }
-  fp = fopen(file_name, "r");
+  fp = fopen(fp_filepath, "r");
   if (fp == 0) {
-    putstr("Could not open file: "); putstr(file_name); putchar('\n');
+    putstr("Could not open file: "); putstr(fp_filepath); putchar('\n');
     exit(1);
   }
+  return fp;
+}
+
+void include_file(char *file_name, char *relative_to) {
+  fp = fopen_source_file(file_name, relative_to);
```

Comment:
It would be nice if we could pass the result of `str_concat(relative_to, file_name)` without using a global variable but `&` cannot be used on in `pnut-sh` so there's no other option.